src/sync/app.cpp

```diff
@@ -622,5 +622,53 @@ void App::log_out(std::function<void (Optional<AppError>)> completion_block) con
     log_out(current_user(), completion_block);
 }
 
+void App::switch_user(std::shared_ptr<SyncUser> user,
```

Comment:
Why do we log the user out when switching between them? It should be possible to have multiple users logged in at the same time? This would also remove the need for the callback, which feels really awkward for this type of API.

Reply:
we don't `log_out` on `switch_user`, we only perform the log_out on `remove_user`

Reply:
Hmm, could have sworn there was code doing that, but why is there a callback with an Error in that case. That shouldn't be needed?

Reply:
If the user is not logged in an error will be returned in the callback

Reply:
Yes, but why making it a callback based API? That just makes it more annoying for consumers. There is nothing in this method that seems to require it, so I would just throw it as an exception?

Reply:
After talking offline, I've removed the callback and changed to returning a shared pointer to the user, and throwing an exception otherwise